### PR TITLE
Rework docs, fix typings, other fixes

### PR DESCRIPTION
**Pull request by Eejit43**
Changes made in this PR:

– Entirely reworked README and all docs, adding examples
– Add more version comparison directly to the `minecraftData` object (`isOlderThan`, etc.)
– Correct missing typings and related issues
– Add `entities` data (entities sorted by ID)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,19 @@
...
 ## Features
 
-For example it's often useful to :
-
-* find blocks by id
-* find items by name
-* find block or item by name
-* find block or item by id
-
-## Example
```
Comment (rom1504) [inline README.md:20]: Why did you remove that?

Comment (Eejit43) [reply README.md:20]: Just thought it was unnecessary, as it is mentioned that below. Are you specifically referring to the example or the possible uses list?

**Comment by rom1504** [comment]
I'd really advise against this kind of "let's change everything" PR without first talking about the objective

File: bin/generate_data.js [hunk 2]
```diff
@@ -4,32 +4,45 @@ const dataSource = require('../minecraft-data/data/dataPaths')
...
-              // Check if the file can be loaded as JSON
-              require('../' + loc + k3 + '.json')
-              return `      get ${k3} () { return require("./${loc}${k3}.json") }`
-            } catch {
-              // No ? Return it as a URL path so other code can decide how to handle it
-              const file = fs.readdirSync(path.join(__dirname, '../', loc)).find(f => f.startsWith(k3 + '.'))
-              if (file) { return `      ${k3}: __dirname + '/${loc}${file}'` } else { throw Error('file not found: ' + loc + k3) }
-            }
-          })
+const data =
```
[the remark this answers is not in the document]
Comment (Eejit43) [reply bin/generate_data.js:7]: No change there, that's just my formatter, it won't affect the data generation at all

**Comment by Eejit43** [comment]
> I'd really advise against this kind of "let's change everything" PR without first talking about the objective

I'm totally open to reverting some changes if you think it's too much, however the only major thing I changed was the docs, and then just a couple method additions, nothing breaking at all

**Comment by extremeheat** [comment]
Sounds like you should start from a new branch and only do one set of features per PR. Any change to the API breaking or not is a new feature. Update Docs for the feature you want to add/modify, don't change lots of unrelated things in a feature PR.

**Comment by Eejit43** [comment]
> Sounds like you should start from a new branch and only do one set of features per PR. Any change to the API breaking or not is a new feature. Update Docs for the feature you want to add/modify, don't change lots of unrelated things in a feature PR.

Which specific things do you want to be split up? Most of the changes were formatting and docs changes. The other changes are very minor and don't affect any current API stuff except for the `foodsByFoodPoints` and `foodsBySaturation` changes, which I'll go ahead right now and create a separate PR for.

File: index.js [hunk 3]
```diff
@@ -37,38 +47,46 @@ function Version (type, version, majorVersion) {
...
   const mcData = data[majorVersion.type][majorVersion.majorVersion]
-  if (mcData == null) { return null }
+  if (mcData === null) return null
+
   const nmcData = mcDataToNode(mcData)
   nmcData.type = majorVersion.type
-  nmcData.isNewerOrEqualTo = version => nmcData.version['>='](version)
-  nmcData.isOlderThan = version => nmcData.version['<'](version)
+  nmcData.isOlderThan = (version) => nmcData.version['<'](version)
+  nmcData.isOlderOrEqualTo = (version) => nmcData.version['<='](version)
```
Comment (extremeheat) [inline index.js:83]: Change

File: lib/loader.js [hunk 4]
```diff
@@ -39,6 +39,7 @@ function mcDataToNode (mcData) {
 
     mobs: indexes.mobsById,
     objects: indexes.objectsById,
+    entities: indexes.entitiesById,
```
Comment (extremeheat) [inline lib/loader.js:42]: Change

**Review by extremeheat — CHANGES_REQUESTED** [review]
Because you did so much unnecessary changes to the formatting it's hard to tell what's changed. But I pointed some things out that stand out

**Comment by Eejit43** [comment]
> Because you did so much unnecessary changes to the formatting it's hard to tell what's changed. But I pointed some things out that stand out

So you think I need to create a separate PR for those tiny changes? (not mad, genuine question)

**Comment by rom1504** [comment]
I see `+2,011 −1,407`

that's a lot of work to review

**Comment by Eejit43** [comment]
Look I'm really sorry. If I'd thought I would have put it in separate PRs. Please let me know what I can do that hopefully isn't separating everything out.

File: README.md (hunk 1 above)
Comment (rom1504) [reply README.md:20]: Specifically referring to the examples. They're here to help people start faster

File: bin/generate_data.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rom1504) [reply bin/generate_data.js:7]: Ok undo the change then

**Comment by rom1504** [comment]
I think it would be a bad move for the repo to accept this.
Simply redo smaller PRs from scratch.

**Comment by Eejit43** [comment]
split into:

– #206
– #207
– #208
– #209

(and more to be made)
